### ISSUE TITLE
Cannot open at all Premium modules.

### ISSUE THREAD

**Issue by Xehanort88** [Beamdog/nwn-issues#154]
![auroraerror1](https://user-images.githubusercontent.com/30383638/92287776-ab3b8880-ef02-11ea-9aa7-1a40b5216ded.png)
**To Reproduce**

 - This is a sample step list. Replace with yours matching the same pattern.
 - Load any premium module, open first area.
 - Toolset crashes ENDLESSLY, has to be stopped by task manager

**Specifics**

 - Latest game build

As per title, it is impossible to open premium modules. The workaround suggested by moderators on Discord doesn't seem to work.

**Comment by Bill-Harper** [comment]
I'm not sure about the suggested workaround, but after a little work I'm seeing Pirates open fine. Here's what I did:

-  I opened potsc_top.hak, and exported all the resources to a folder

- I renamed Neverwinter Nights - Pirates of the Sword Coast.nwm to pirates.mod (any 16-char/less name the toolset accepts is fine)

- I opened pirates.mod in NWHak, and added all the resources I had exported from potsc_top

- I saved, and placed pirates.mod in my modules folder

And everything opened fine. I realize it's a bit of inconvenience; the original idea was to prevent large save bloats. I'll discuss this with the team, and see if we might want to open these up at least for the PC version.

**Comment by Xehanort88** [comment]
Thank you - I mean, it's a bit silly that all this hassle is required to save some MBs when now HDDs feature TBs of space, isn't it?

**Comment by DMSymphony** [comment]
Well, save space was a huge issue on consoles, especially switch, where both the game's install and data has to fit on an SD card.

Because of the way that the game saves work, not doing things this way would be the same as saving your module as a new name every time you feel like saving in the toolset. All module files are duplicated every game save, so, putting only the files that CAN change (not the area's, blueprints, or scripts, etc) in the module dramatically lessens the size of each of those saves.

As far as some MB's is concerned, 3 saves is just about 1/10th a GB. -I- Have like 14 TB or something, and I know Bill has a lot too, but we're on desktop PC's. It's very common for me to talk to NWN players who tell me they have their game on SSD, and it's 120 GB and 90% full, which is bad for an SSD. Many of them install their Steam games to auxillary HDD's, which is unfortunate, considering NWN's base install is only 6 GB or so, but that won't protect them from save bloat unless they move their Documents folder off C:, which doesn't seem to be common knowledge.

It's been considered to have optional downloads of .mod versions of the premium modules, not only with a normal module setup, but also with things like TGA textures and ascii models, etc, but it's unfortunately low priority. I'm sorry for the hassle, but if I was going to compare hassles between all of the people playing the Premiums and all of the people opening them in the toolset, I'm going to side with the player's first, because, there's obviously way more of them, and, someone opening the PM in the toolset probably has some interesting ideas in mind, and is more likely to be able to overcome tricks like this to begin with.

If it makes you feel better, before December they were encrypted and you couldn't open them at all.

**Comment by Xehanort88** [comment]
I'm well aware DMSymphony, I made the request more than 2 years ago...

**Comment by Xehanort88** [comment]
> I'm not sure about the suggested workaround, but after a little work I'm seeing Pirates open fine. Here's what I did:
> 
> * I opened potsc_top.hak, and exported all the resources to a folder
> * I renamed Neverwinter Nights - Pirates of the Sword Coast.nwm to pirates.mod (any 16-char/less name the toolset accepts is fine)
> * I opened pirates.mod in NWHak, and added all the resources I had exported from potsc_top
> * I saved, and placed pirates.mod in my modules folder
> 
> And everything opened fine. I realize it's a bit of inconvenience; the original idea was to prevent large save bloats. I'll discuss this with the team, and see if we might want to open these up at least for the PC version.

I did that and it opened fine. However, no changes I make work. Is it because of the attached hak pak, which I could perhaps remove?

**Comment by DMSymphony** [comment]
If you take the files out of the hak and put them in the module, you should take them out of the hak. You have read-only duplicates that are prioritized first, You will absolutely get messages like the above.

You can't open the module without the haks until you remove them (.mod requires named .haks), and it's not easy to remove the haks without opening the module.

What may have been unclear in the instructions above is that you shouldn't leave the area resources in the haks. Find or make a tiny .txt file, with nothing in it if you want, to be the only thing left in the module hak, and save that empty version to /hak. You can't save an empty hak, so you need the .txt in there.

Once you get the toolset to open, open module properties to remove the module resources hak from the module, then you will only leave in the un-altered custom content haks (tilesets, etc.)

**Comment by Xehanort88** [comment]
> If you take the files out of the hak and put them in the module, you should take them out of the hak. You have read-only duplicates that are prioritized first, You will absolutely get messages like the above.
> 
> You can't open the module without the haks until you remove them (.mod requires named .haks), and it's not easy to remove the haks without opening the module.
> 
> What may have been unclear in the instructions above is that you shouldn't leave the area resources in the haks. Find or make a tiny .txt file, with nothing in it if you want, to be the only thing left in the module hak, and save that empty version to /hak. You can't save an empty hak, so you need the .txt in there.
> 
> Once you get the toolset to open, open module properties to remove the module resources hak from the module, then you will only leave in the un-altered custom content haks (tilesets, etc.)

I actually don't empty the hak, I add the resources to the module and delete it afterwards from within the editor. What about the inconsistent entries, however?

**Comment by Xehanort88** [comment]
![auroraerror5](https://user-images.githubusercontent.com/30383638/93709025-dcd36700-fb32-11ea-8b0e-0b10f4c555b1.png)

Yet another resource error when opening the module according to the method you suggested.

**Comment by virusman** [comment]
Re: original issue, this is because the module only contains .git files for the areas, the .are files are in the hak, and the Toolset requires all area files to be present in the working directory (the module).

**Comment by virusman** [comment]
It does make sense I think; I'd consider it more of a content issue (missing .are files in the module) rather than a toolset issue.
The only thing that's incorrect is that Toolset should show an error and not try to open the area.
